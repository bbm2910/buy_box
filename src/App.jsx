import './App.css';
import { useState } from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Navbar from './components/Navbar';
import HomePage from './components/HomePage';
import AboutPage from './components/AboutPage';
import Footer from './components/Footer';
import ShopPage from './components/ShopPage';

import { CartProvider } from './components/CartContext';

const App = () => {
	const [showCart, setShowCart] = useState(false);

	return (
		<Router>
			<CartProvider>
				<div>
					<Navbar setShowCart={setShowCart} />

					<Routes>
						<Route
							path='/shop'
							element={
								<ShopPage
									showCart={showCart}
									setShowCart={setShowCart}
								/>
							}
						/>
						<Route
							path='/'
							element={<HomePage showCart={showCart} />}
						/>
						<Route
							path='/about'
							element={<AboutPage showCart={showCart} />}
						/>
					</Routes>
				</div>
			</CartProvider>
			<Footer />
		</Router>
	);
};

export default App;
