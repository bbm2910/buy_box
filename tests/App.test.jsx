import { render } from '@testing-library/react';
import App from '../src/App';

describe('App component', () => {
	it('renders without crashing', () => {
		const { container } = render(<App />);
		expect(container).toMatchSnapshot();
	});
});
