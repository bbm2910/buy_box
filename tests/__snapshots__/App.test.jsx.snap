// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders without crashing 1`] = `
<div>
  <div>
    <nav>
      <div
        class="links"
      >
        <a
          href="/"
        >
          Homepage
        </a>
        <a
          href="/shop"
        >
          Shop
        </a>
        <a
          href="/about"
        >
          About
        </a>
      </div>
      <a
        class="cart-btn"
        href="/shop"
      >
        <img
          alt=""
          src="/src/assets/shopping-cart.svg"
        />
        <p>
          0
        </p>
      </a>
    </nav>
    <div
      class="homepage"
    >
      <h2
        id="text1"
      >
        Unbeatable Prices
      </h2>
      <h2
        id="text2"
      >
        Doorbuster Price
      </h2>
      <h2
        id="text3"
      >
        Exclusive Promotions
      </h2>
      <h2
        id="text4"
      >
        Markdown Madness
      </h2>
      <h2
        id="text5"
      >
        Clearance Event
      </h2>
      <h2
        id="text6"
      >
        Limited-Time Offer
      </h2>
      <h2
        id="text7"
      >
        Mega Salet
      </h2>
    </div>
  </div>
  <footer>
    Â© BM 2024
  </footer>
</div>
`;
